src/main/java/com/amplitude/Retry.java

```diff
@@ -176,7 +178,7 @@ private static void retryEventsOnLoop(String userId, String deviceId, HttpCall h
                   if (shouldReduceEventCount && !isLastTry) {
                     eventCount /= 2;
                   }
-                } catch (InterruptedException e) {
+                } catch (InterruptedException | AmplitudeInvalidAPIKeyException e) {
```

Comment:
Same thing I guess we are not really getting this exception here. Also by catching the exception in there, would the purpose is we keep trying.